tests/mcpdotnet.TestServer/Program.cs

```diff
@@ -53,7 +55,19 @@ private static async Task Main(string[] args)
         Log.Logger.Information("Server started.");
 
         // Run until process is stopped by the client (parent process)
-        await Task.Delay(Timeout.Infinite);
+        while (true)
+        {
+            await Task.Delay(5000);
+            foreach (var resource in _subscribedResources)
```

Comment:
Seems like there's a possible concurrency bug here... this might be enumerating the resources concurrently with the SubscribeToResourcesHandler delegate below mutating the collection?